Prop usage is not detected

The following code:

```jsx
const React = require('react')
const someArray = require('./someArray')

const Test = props => <div>
    {someArray.map(l => <div
      key={l}>
        {props.property + props.property2}
      </div>)}
  </div>

Test.propTypes = {
  property: React.propTypes.string.isRequired,
  property2: React.propTypes.string.isRequired
}
```
raises error 

```
  12:13  error  'property' PropType is defined but prop is never used  react/no-unused-prop-types
````

with eslint-plugin-react version 6.10.0.

--- a/tests/lib/rules/no-unused-prop-types.js
+++ b/tests/lib/rules/no-unused-prop-types.js
@@ -1914,6 +1914,21 @@ ruleTester.run('no-unused-prop-types', rule, {
         '    test: React.PropTypes.bool',
         '};'
       ].join('\n')
+    }, {
+      // issue 1107
+      code: [
+        'const Test = props => <div>',
+        '  {someArray.map(l => <div',
+        '    key={l}>',
+        '      {props.property + props.property2}',
+        '    </div>)}',
+        '</div>',
+
+        'Test.propTypes = {',
+        '  property: React.propTypes.string.isRequired,',
+        '  property2: React.propTypes.string.isRequired',
+        '}'
+      ].join('\n')
     }
   ],
 

